usr/sbin/p11sak/p11sak.c

```diff
@@ -912,6 +927,55 @@ static void set_bool_attr_from_string(CK_ATTRIBUTE *attr, char attr_char)
     }
 }
 
+/**
+ * @brief Set the custom attrs from string object
+ * 
+ * @param attr The attribute array
+ * @param attr_char The string from the user
+ * @param count Index of current attribute in the attribute array
+ */
+static CK_RV set_custom_attrs_from_string(char *attr_char, CK_ATTRIBUTE *attr, CK_ULONG *count) 
+{
+    CK_RV rc = CKR_OK;
+    int i = 0, j = 0;
+    char substr[10];
+
+    // iterate over the single custom attributes
+    while (i < (int)strlen(attr_char))
+    {
+        int radix = 16;
+        unsigned long id = 0;
+
+        // parse attribute id from string
+        for (j = 0; attr_char[i] != '='; i++) {
```

Comment:
you could use strtok() to iterate over the token(s) instead of parsing t yourself. But that would mean a different logic of this routine. And it would modify attr_char....

Reply:
That is a weird-looking loop: Initialize `j`, but use and increment `i`.  At a quick glance it looks like a typo but it does not seem to be...

Reply:
Done